src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -227,6 +236,9 @@ function setInitialDOMProperties(
       // Noop
     } else if (registrationNameModules.hasOwnProperty(propKey)) {
       if (nextProp) {
+        if (__DEV__) {
+          warnForInvalidEventListener(propKey, nextProp);
```

Comment:
This is a very hot path. How do you feel about hoisting the condition right here (duplicating it) and only calling the function in the bad case?